### PR TITLE
Use GitHub Actions to run tests and publish

### PR DESCRIPTION
**Pull request by AndyBitz**
- Moves to GitHub Actions for tests and publishing to be more aligned with our other repos.
 - Replaces the Codecov dependency with its GitHub Action.

Will remove Circle and change the required action once this PR got approved.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Tests
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
[the remark this answers is not in the document]
Comment (AndyBitz) [reply .github/workflows/tests.yaml:7]: 👍 Yes, planned on doing that!

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Publish Package
+
+on:
+  release:
+    types: [created]
```
Comment (styfle) [inline .github/workflows/publish.yaml:5]: How does this get invoked? Do we need to run `npm version` somewhere?

**Review by styfle — APPROVED** [review]
Thanks!

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (AndyBitz) [reply .github/workflows/publish.yaml:5]: As far as I understand, when a new release gets created on https://github.com/vercel/serve-handler/releases

So, we'd continue to run  e.g. `release` to create a new version + update package.json + prepare the release. And once the release gets published/created, that action would run.

Comment (styfle) [reply .github/workflows/publish.yaml:5]: I see, so there will still be a manual process and committing/pushing to main directly.

In the future, we might want to automate that part too 👍

Comment (kachkaev) [reply .github/workflows/publish.yaml:5]: We’re using https://github.com/changesets/changesets in [blockprotocol/blockprotocol](https://github.com/blockprotocol/blockprotocol), works pretty well for managing changelogs & releases on GitHub!

**Pull request merged**